Update README.md with Pagination (#362)
woylie/flop_phoenix: README.md

in addition to a path. For more details, please refer to the component
documentation.

> To change page size, you can either add a HTML `input` control (see below) or

Reviewer: ```suggestion
To change the page size, you can either add an HTML `input` control (see below) or
```

add a LiveView `handle_event/3` function with a corresponding control, such 
as a link. For example, you might create a widget with several page-size links. 

> You could create a page_size_link component, like this: 

Reviewer: ```suggestion
You could create a `page_size_link` component like this: 
```

```elixir
attr :current_size, :integer, required: true
attr :new_size, :integer, required: true

def page_size_link(assigns) do
  ~H"""
  <.link
    phx-click="page-size"
    phx-value-size={@new_size}
    class={page_size_class(@current_size, @new_size)}
  >
    <%= @new_size %>
  </.link>
  """
end

defp page_size_class(old, old), do: "font-black text-orange-500"
defp page_size_class(_old, _new), do: "font-light"
```

You could then render them like this: 

```elixir

> <%= for ps <- [10, 20, 40, 60] do %>
>   <.page_size_link new_size={ps} current_size={@meta.page_size} />
> <% end %>

Reviewer: ```suggestion
<.page_size_link :for={ps <- [10, 20, 40, 60]} new_size={ps} current_size={@meta.page_size} />
```

```

Then, you can handle the event in your LiveView, like this: 

```elixir
  def handle_event("page-size", %{"size" => ps}, socket) do
    flop = %{socket.assigns.meta.flop | page_size: ps, limit: nil}
    path = Flop.Phoenix.build_path(~p"/pets", flop)

    {:noreply, push_patch(socket, to: path)}
  end
```

> This method allows you to update page size while maintaining browser history. 

Reviewer: ```suggestion
This method allows you to update the page size while maintaining browser history. 
```
